dali/test/python/test_operator_one_hot.py

```diff
@@ -11,34 +11,74 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from functools import partial
 
 from nvidia.dali.pipeline import Pipeline
 import nvidia.dali.ops as ops
 import nvidia.dali.types as types
 from test_utils import check_batch
 
+from nose.tools import raises
+
 import numpy as np
 
+
 num_classes = 20
 batch_size = 10
 
 
 def insert_as_axis(target, value, axis, total_axes):
     return target[0:axis] + (value,) + target[axis:total_axes]
 
+
+def get_initial_layout(sample_dim=0, insert_at=None, new_axis_name='O', base="ABCD"):
+    return base[0:sample_dim]
+
+
+def modify_layout(layout, output_dim, axis=None, layout_axis_name='O'):
+    if layout_axis_name is not None:
+        layout_axis_name = str(layout_axis_name)
+    if not layout_axis_name or (not layout and output_dim > 1):
+        return ""
+    if axis is None:
+        return layout_axis_name
+    if axis < 0:
+        axis = len(layout)
+    return layout[:axis] + layout_axis_name + layout[axis:]
+
+
+def random_3d_tensors_batch():
+    return [
+        np.random.randint(0, num_classes, size=np.random.randint(2, 8, size=(3,)), dtype=np.int32)
+        for _ in range(batch_size)
+    ]
+
+
+def random_scalars_batch():
+    return np.random.randint(0, num_classes, size=batch_size, dtype=np.int32)
+
+
+def random_scalar_like_tensors_batch(nested_level):
+    return [
+        np.array([np.random.randint(0, num_classes)], dtype=np.int32).reshape((1, ) * nested_level)
+        for x in range(batch_size)
+    ]
+
+
 class OneHotPipeline(Pipeline):
-    def __init__(self, num_classes, input, axis=-1, num_threads=1):
+    def __init__(self, num_classes, input, axis=-1, num_threads=1, layout=None, layout_axis_name="O"):
         super(OneHotPipeline, self).__init__(batch_size,
                                              num_threads,
                                              0)
-        sample_dim = len(input[0].shape)
-        self.ext_src = ops.ExternalSource(source=[input], cycle=True, layout="ABCD"[0:sample_dim])
-        self.one_hot = ops.OneHot(num_classes=num_classes, axis=axis, dtype=types.INT32, device="cpu")
+        self.ext_src = ops.ExternalSource(source=[input], cycle=True, layout=layout)
```

Comment:
I know this is a bit out of scope here, but can we maybe move the batch generation to here (pass the callbacks that you already created) - so we have several different inputs when iterating.